### PR TITLE
It's 2017, time to fuck (some) spawns and cleanbots

### LINKED ISSUE #14893: Sound procs are being replaced by New() calls, breaking everything (CLEANBOT, corgis, etc)
!!!SEE COMMENTS FOR RELEVANT INFO!!!

1) A global narrate shows up for players spontaneously, often after announcements (NOTE FROM THE FUTURE: because they play sounds). Here is what it looks like (the first announcement is from a silicon who happened to get named Cleanbot after this nonsense started):

Cleanbot Announces:
I am in your head now. Cleanbot.

What players see immediately after: (cleanbot minicon) Cleanbot

2) Over 6000 cleanbots spawn in nullspace. Server crashes, either on its own or due to the efforts of admins trying to del-all 6000 cleanbots. rip

img

from runtime logs the Cleanbot (/mob/living/simple_animal/bot/cleanbot): New(null, null, null, null, null) 

(/mob/living/simple_animal/bot/cleanbot): New('sound/ambience/title1.ogg', null, null, null, null)  -- relatedly all in-game sounds stopped working for everyone around now, admin did disable playing sounds at around the same time though

### PR DESCRIPTION
**Pull request by Cyberboss**
I'm carefully reviewing each of these to pick the best course (addtimer, waitfor = 0)

Removed check_for_cleanbot_bug and closes #14893, it's been gone in 511

### CODE REVIEW

File: code/_onclick/telekinesis.dm [hunk 1]
```diff
@@ -179,9 +179,7 @@ var/const/tk_maxrange = 15
 	O.icon = 'icons/effects/effects.dmi'
 	O.icon_state = "nothing"
 	flick("empdisable",O)
-	spawn(5)
-		qdel(O)
-
+	QDEL_IN(O, 5)
```
Comment (ChangelingRain) [inline code/_onclick/telekinesis.dm]: wait why isn't this just an `overlay/temp` I'm gonna Fix That

**Comment by Mervill** [comment]
what if I only like `spawn()` as a friend?

**Comment by Cyberboss** [comment]
It's been a long time for spawn though

**Comment by CPTANT** [comment]
For us noobs, what is the difference between sleep and spawn?

**Comment by Supermichael777** [comment]
sleep is awful use spawn
spawn is bad use the timer function

**Comment by XDTM** [comment]
sleep pauses the code, it will wait the indicated time before proceeding.
spawn runs the code tabbed under it after the indicated time, but it won't stop running the rest of the code in the meantime.

```
do 1
sleep(5)
do2
do3
```

will cause do1 ... [5 seconds pass] ... do2 do3

while

```
do1
spawn(5)
    do2
do3
```

will cause do1 do3 ... [5 seconds] ... do2

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1312,31 +1313,37 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
...
 
 #define QDEL_IN(item, time) addtimer(CALLBACK(GLOBAL_PROC, /proc/qdel, item), time)
 
 /proc/check_for_cleanbot_bug()
+	set waitfor = 0
 	var/static/admins_warned //bet you didn't know you could do this!
 	var/icon/Icon_test = icon('icons/BadAss.dmi')
+	var/msg1 = null
+	var/msg2 = null
 	if(!istype(Icon_test))
```
Comment (MrStonedOne) [inline code/__HELPERS/unsorted.dm]: just no. remove this proc entirely =P

Comment (Cyberboss) [inline code/__HELPERS/unsorted.dm]: finally

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1312,32 +1313,12 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 		return
 
 	C.color = flash_color
-	spawn(0)
-		animate(C, color = initial(C.color), time = flash_time)
+	animate(C, color = initial(C.color), time = flash_time)
```
Comment (MrStonedOne) [inline code/__HELPERS/unsorted.dm]: animate doesn't sleep.

Comment (Cyberboss) [inline code/__HELPERS/unsorted.dm]: Removed the waitfor = 0

**Comment by Cyberboss** [comment]
@optimumtact Reopen, I'm done with this. I'll get more later

File: code/controllers/subsystem/assets.dm [hunk 4]
```diff
@@ -15,7 +15,10 @@ var/datum/subsystem/assets/SSasset
...
 	for(var/client/C in clients)
-		// Doing this to a client too soon after they've connected can cause issues, also the proc we call sleeps.
-		spawn(10)
-			getFilesSlow(C, cache, FALSE)
+		GiveFilesToClient(C)
 	..()
+
+/datum/subsystem/assets/proc/GiveFilesToClient(client/C)
+	set waitfor = 0
+	sleep(10)	// Doing this to a client too soon after they've connected can cause issues, also the proc we call sleeps.
```
Comment (MrStonedOne) [inline code/controllers/subsystem/assets.dm]: Addtimer is faster then sleep, so making this use addtimer rather then set waitfor = 0 then sleep would be better

File: code/_onclick/hud/fullscreen.dm [hunk 5]
```diff
@@ -23,19 +23,19 @@
...
 
 	if(animated)
-		spawn(0)
-			animate(screen, alpha = 0, time = animated)
-			sleep(animated)
-			if(client)
-				client.screen -= screen
-			qdel(screen)
+		animate(screen, alpha = 0, time = animated)
+		sleep(animated)
```
Comment (MrStonedOne) [inline code/_onclick/hud/fullscreen.dm]: Should use a clienttime timer since animate() runs on client time and not byond time.

**Pull request merged**